core/templates/services/playthrough.service.ts

```diff
@@ -114,12 +117,15 @@ class CyclicStateTransitionsTracker {
 }
 
 class EarlyQuitTracker {
-  private stateName: string = null;
-  private expDurationInSecs: number = null;
+  // This property is initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  private stateName!: string;
+  private expDurationInSecs!: number;
 
   foundAnIssue(): boolean {
     return (
-      this.expDurationInSecs !== null &&
+      this.expDurationInSecs !== undefined &&
```

Comment:
Is it guaranteed that, when this class is initialized, expDurationInSecs has a valid numeric value?

If so, do we need a TODO to remove this check when the codebase is fully typed?

Reply:
Done